components/log-viewer-webui/client/src/api/socket/MongoCollectionSocket.ts

```diff
@@ -0,0 +1,50 @@
+import {
+    ClientToServerEvents,
+    ServerToClientEvents,
+} from "@common/index.js";
+import {Socket} from "socket.io-client";
+
+import {MongoCursorSocket} from "./MongoCursorSocket.js";
+import {getSharedSocket} from "./SocketSingleton.js";
+
+
+/**
+ * Socket connection to a MongoDB collection residing on a server. Class provides methods to
+ * query the collection.
+ */
+class MongoCollectionSocket {
+    collectionName: string;
```

Comment:
sorry for missing this in the last batch of comments

does it need to be public? if not, let's change it to private